pkg/kubectl/cmd/util/factory.go

```diff
@@ -83,6 +83,8 @@ type Factory struct {
 	DefaultNamespace func() (string, bool, error)
 	// Returns the generator for the provided generator name
 	Generator func(name string) (kubectl.Generator, bool)
+	// Check whether the kind of resources could be exposed
+	CanBeExposed func(kind string) error
```

Comment:
@smarterclayton any strong preference between `func(kind string)` and `func(obj runtime.Object)`?